Automattic/jetpack
# Connection: show full screen connection banner modal in JP dash

Fixes #10563
Fixes #8618

#### Changes proposed in this Pull Request:

That modal was only loaded in the Plugins screen; let's load it in the Jetpack menu as well.

#### Notes

Since we are getting rid of the old connection component that used to be displayed on the Jetpack dashboard page, we now have to reload the page when you disconnect your site from WordPress.com, so the new modal can appear again.

By doing so, we do not see the notice that appears at the top of the Jetpack dashboard after a disconnection, inviting you to take a survey to tell us why you disconnected.

<img width="890" alt="screenshot 2018-12-10 at 16 49 07" src="https://user-images.githubusercontent.com/426388/49743765-c3ce2e80-fc9b-11e8-9a29-e875cbd8a560.png">

#### Testing instructions:

* Start from a site where Jetpack is not active.
* Activate the plugin.
* Check the connection modal appearing after the plugin is activated.
* Go to the Jetpack menu. You should see the same modal.

#### Proposed changelog entry for your changes:

* Improve the connection flow for new Jetpack users.
